packages/instantsearch-react/src/connectors/connectSearchBox.js

```diff
@@ -37,9 +37,16 @@ export default createConnector({
     return searchParameters.setQuery(getQuery(props, state));
   },
 
-  getMetadata(props) {
+  getMetadata(props, state) {
+    const query = getQuery(props, state);
     return {
       id: props.id,
+      filters: !query ? [] : [{
+        key: props.id,
+        label: query,
+        hide: true,
```

Comment:
It could be, it's just a matter of taking `props` into account here. We could also decide on the `CurrentFilters` side by passing it the `ids` of the filters we want to show. I went with this as a default because I wasn't sure of what we wanted to achieve with this widget, but this is a possibility for sure.